src/pages/en/core-concepts/routing.md

```diff
@@ -187,6 +149,55 @@ const { title, text } = Astro.props;
 </html>
 ```
 
+### Server (SSR) Mode
+In [SSR mode](/en/guides/server-side-rendering/), dynamic routes are defined the same way: include `[param]` or `[...path]` brackets in your file names to match arbitrary strings or paths. But because the routes are no longer built ahead of time, the page will be served to any matching route, and `getStaticPaths` has no effect. 
+
+```astro title="src/pages/resources/[resource]/[id].astro"
+---
+const { resource, id } = Astro.params;
+---
+<h1>{resource}: {id}<h1>
+```
+This page will be served for any value of `resource` and `id`: `resources/users/1`, `resources/colors/blue`, etc.
+
+#### Modifying the `[...slug]` example for SSR
+
+Because SSR pages can't use `getStaticPaths`, they can't receive props. Here, we modify our [earlier example](#example-dynamic-pages-at-multiple-levels) to work in SSR by looking up the value of the `slug` param in an object.
+
+```astro title="src/pages/[...slug].astro"
+---
+const pages = {
+  "products": {
+    title: "Astro products",
+    text: "We have lots of products for you",
+  },
+  "products/astro-handbook": {
+    title: "The ultimative Astro handbook",
+    text: "If you want to learn Astro, you must read this book.",
+  },
+}
+
+const defaultPage = {
```

Comment:
Instead of having a default, would it make sense to show a 404 example here if the page isn't found?

Reply:
Ah thanks for pointing this out, this example actually isn't equivalent to the SSG one because of this - changed!